tests/test_parse/test_genbank.py

```diff
@@ -552,7 +536,46 @@ def test_rich_parser_info(rich_gb):
 
 def test_rich_genbank_just_seq():
     with open("data/annotated_seq.gb") as infile:
-        parser = RichGenbankParser(infile, just_seq=True)
+        parser = genbank.rich_parser(infile, just_seq=True)
         seq = [s for l, s in parser][0]
 
     assert not len(seq.annotation_db)
+
+
+def test_iter_genbank_records(DATA_DIR):
+    path = DATA_DIR / "annotated_seq.gb"
+    name, seq, features = list(genbank.iter_genbank_records(path))[0]
+    assert len(seq) == 6201
+    assert name == "AE017341"
+    assert seq.startswith("CAATACCCAC")
+    assert seq.endswith("TGTGTACGTAA")
+    assert features["locus"] == "AE017341"
+
+
+def test_iter_genbank_records_no_feature_conversion(DATA_DIR):
+    path = DATA_DIR / "annotated_seq.gb"
+    name, seq, features = list(
+        genbank.iter_genbank_records(path, convert_features=None)
+    )[0]
+    assert len(seq) == 6201
+    assert name == "AE017341"
+    assert seq.startswith("CAATACCCAC")
+    assert seq.endswith("TGTGTACGTAA")
+    assert isinstance(features, str)
+
+
+@pytest.fixture()
+def no_seq(DATA_DIR, tmp_path):
+    seqfile = DATA_DIR / "annotated_seq.gb"
+    data = seqfile.read_text()
+    index = data.find("\nORIGIN")
+    out_seqfile = tmp_path / "noseq.gb"
+    out_seqfile.write_text(f"{data[:index]}\nORIGIN\n//\n")
+    return out_seqfile
+
+
+def test_iter_genbank_records_noseq(no_seq):
+    name, seq, features = list(genbank.iter_genbank_records(no_seq))[0]
```

Comment:
features is not used, but IMO it's not worth renaming it with `_` to fix the codacy warning.  `name, seq, features` is much more descriptive.